shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm

```diff
@@ -89,14 +101,19 @@
     // In case the application bundle is still not specified, look for the App.framework in the
     // Frameworks directory.
     if (settings.application_library_path.size() == 0) {
-      NSString* applicationFrameworkPath = [mainBundle pathForResource:@"Frameworks/App.framework"
-                                                                ofType:@""];
-      if (applicationFrameworkPath.length > 0) {
-        NSString* executablePath =
-            [NSBundle bundleWithPath:applicationFrameworkPath].executablePath;
-        if (executablePath.length > 0) {
-          settings.application_library_path.push_back(executablePath.UTF8String);
-        }
+      NSString* libraryPath = FindFrameworkApplicationLibraryPath(mainBundle);
+      if (libraryPath.length > 0) {
+        settings.application_library_path.push_back(libraryPath.UTF8String);
+      }
+    }
+  }
+
+  // Check other bundles if we still haven't found anything.  This is useful for XCTest targets.
+  if (settings.application_library_path.size() == 0) {
+    for (NSBundle* bundle in [NSBundle allBundles]) {
```

Comment:
Shouldn't App.framework be one of these bundles, so check for its existence as bundle in the loop instead of checking the Framework directories?  Or is the issue that the test can't load the symbols, but it's still embedded?  Or are there multiple of them, which is why you don't break from the loop when you find the first?  Sorry, I'm not familiar with this code... I would have expected `bundle = [NSBundle bundleWithIdentifier:[FlutterDartProject defaultBundleIdentifier]];` to be the App.framework, even in the tests.

Also nit allBundles is a class method, so dot notation `NSBundle.allBundles`

Reply:
I explain why I was doing this in a previous comment to xster.  I've since removed the code so it's moot now.  TLDR App.framework isn't one of the bundles, the .xctest is the bundle that has App.framework in its Frameworks.  That's how we do it elsewhere.